app/services/order_appender.rb

```diff
@@ -0,0 +1,71 @@
+class OrderAppender
+
+  attr_reader :fulfilled_at, :note, :order_status_id, :original_order, :product,
+              :quantity, :user
+
+  def initialize(product:, quantity:, original_order:, user:, note: nil, fulfilled_at: nil, order_status_id: nil)
+    @product = product
+    @quantity = quantity
+    @original_order = original_order
+    @user = user
+    @note = note
+    @fulfilled_at = fulfilled_at
+    @order_status_id = order_status_id
+  end
+
+  def add!
+    notifications = false
+    order_details.each do |order_detail|
+      order_detail.note = note if note.present?
+      order_detail.fulfilled_at = fulfilled_at if fulfilled_at.present?
```

Comment:
I think this needs to happen after the `change_status!`. If I set a date and Complete, the fulfilled at is being set to now.

And this should not happen if the status is not complete (or a substatus of complete). Right now, if I put a date in, and leave status blank, it gets created as New, but with a fulfilled_at.